src/sentry/static/sentry/app/views/organizationGroupDetails/groupEventDetailsContainer.tsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+
+import {fetchOrganizationEnvironments} from 'app/actionCreators/environments';
+import {t} from 'app/locale';
+import {GroupEventDetails} from 'app/views/organizationGroupDetails/groupEventDetails';
+import LoadingError from 'app/components/loadingError';
+import LoadingIndicator from 'app/components/loadingIndicator';
+import OrganizationEnvironmentsStore from 'app/stores/organizationEnvironmentsStore';
+import {Client} from 'app/api';
+import {GlobalSelection, Organization, Environment} from 'app/types';
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  selection: GlobalSelection;
+};
+
+type State = {
+  environments: Environment[];
+  error: Error;
+};
+
+export default class GroupEventDetailsContainer extends React.Component<Props, State> {
+  state = OrganizationEnvironmentsStore.get();
+
+  componentDidMount() {
+    this.environmentSubscription = OrganizationEnvironmentsStore.listen(data =>
+      this.setState(data)
+    );
+    const {environments, error} = OrganizationEnvironmentsStore.get();
+    if (!environments && !error) {
+      fetchOrganizationEnvironments(this.props.api, this.props.organization.slug);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.environmentSubscription) {
+      this.environmentSubscription.unsubscribe();
+    }
+  }
+
+  environmentSubscription: any;
```

Comment:
wasn't sure how to handle the subscription correctly, is there a reflux data type for this?

Reply:
I would just add `TODO(ts): reflux :(`